build/l10n/l10n.js

```diff
@@ -81,21 +81,32 @@
 
   /* API for webapp-optimize */
 
+  // on buildtime, all known pseudolocales need to run through the qps logic;
+  // on runtime this is not the case because some of them might have been
+  // already built on buildtime
+  L10n.Locale.prototype.isPseudo = function() {
+    return this.id in navigator.mozL10n.qps;
+  };
+
   L10n.Locale.prototype.addAST = function(ast) {
     if (!this.astById) {
       this.astById = Object.create(null);
     }
 
+    var isPseudo = this.isPseudo();
+
     /* jshint boss:true */
     for (var i = 0, node; node = ast[i]; i++) {
```

Comment:
I'd like to silence boss warning by surrounding the assignment with parenthesis like:
```for (var i = 0, node; (node = ast[i]); i++) {```